file-sharing-back/file-sharing-back-core/src/main/java/com/file/sharing/core/business/impl/ItemsActionBusinessImpl.java

```diff
@@ -116,6 +114,8 @@ public void saveFileItemAction(Integer itemId, ItemActionType actionType) {
 	}
 
 	@Override
+    @Transactional
+	@CacheEvict(value = "storageInfo", allEntries = true)
```

Comment:
It's not clean to add the cache evict annotation here.
Check my comment also on CacheContextProvider

It's more clear to keep the cache eviction separate from the logic. I suppose this method is called inside a transactional service. So the best place to evict the cache, is when the transaction succeeds. In this case, we should "hook" the cache eviction there. I don't know if we send a spring event when we succeed an item action transaction, but it would be a good idea to do so :)

Reply:
The event is already sent from the UploadFileActionHandler class where the saveFileItem() method is called. 
Did you want the event sending operation to be linked to this method? Like creating an annotation that would wrap this method and send the event automatically?